examples/src/main/scala/com/cloudera/livy/apps/WordCountApp.scala

```diff
@@ -0,0 +1,196 @@
+      if (targetFolderFiles == null || targetFolderFiles.size == 0) {
+        throw new FileNotFoundException("Examples App jar not found in the target folder")
+      }
+      targetFolderFiles.foreach { file =>
+        if (!file.isDirectory && file.getAbsolutePath.contains("jar")) {
+          path = file.getAbsolutePath
+        }
+      }
+    }
+    if (path == "" || !path.contains("jar")) {
+      throw new FileNotFoundException("Jar not found")
+    }
+    path
+  }
+
+  private def uploadJar(path: String) = {
+    val file = new File(path)
+    val uploadJarFuture = scalaClient.uploadJar(file)
+    Await.ready(uploadJarFuture, 40 second) onComplete {
+      case Success(t) => logger.info("Successfully uploaded " + file.getName)
+      case Failure(e) => throw e
+    }
+  }
+
+  /**
+    * Submits a spark streaming job to the livy server
+    * <p/>
+    * The streaming job reads data from the given host and port. The data read
+    * is saved in json format as data frames in the given output path. For simplicity,
+    * the number of streaming batches are 2 with each batch for 20 seconds. The Timeout
+    * of the streaming job is set to 40 seconds
+    * @param host Hostname that Spark Streaming context has to connect for receiving data
+    * @param port Port that Spark Streaming context has to connect for receiving data
+    * @param outputPath Output path to save the processed data read by the Spark Streaming context
+    */
+  def processStreamingWordCount(host: String, port: Int, outputPath: String): ScalaJobHandle[Unit] = {
+    val handle = scalaClient.submit { context =>
+      context.createStreamingContext(20000)
+      val ssc = context.streamingctx
+      val sqlctx = context.sqlctx
+      val lines = ssc.socketTextStream(host, port, StorageLevel.MEMORY_AND_DISK_SER)
+      val words = lines.filter(filterEmptyConent(_)).flatMap(tokenize(_))
+      words.print()
+      words.foreachRDD { rdd =>
+        import sqlctx.implicits._
+        val df = rdd.toDF("word")
+        df.write.mode("append").json(outputPath)
+      }
+      ssc.start()
+      ssc.awaitTerminationOrTimeout(35000)
+      ssc.stop(false, true)
+    }
+    handle
+  }
+
+  /**
+    * Submits a spark sql job to the livy server
+    * <p/>
+    * The sql context job reads data frames from the given json path and executes
+    * a sql query to get the word with max count on the temp table created with data frames
+    * @param inputPath Input path to the json data containing the words
+    */
+  def getWordWithMostCount(inputPath: String): ScalaJobHandle[Any] = {
+    val handle = scalaClient.submit { context =>
+      val sqlctx = context.sqlctx
+      val rdd = sqlctx.read.json(inputPath)
+      rdd.registerTempTable("words")
+      val result = sqlctx.sql("select word, count(word) as word_count from words " +
+        "group by word order by word_count desc limit 1")
+      if (result != null && result.count() > 0) {
```

Comment:
This is not very efficient, since you're running 2 jobs (count + collect).

You should be able to get away with just `result.first`, although I'm not sure how that behaves when there are no rows.